CMakeLists.txt

```diff
@@ -3,20 +3,49 @@ cmake_minimum_required(VERSION 2.6.4)
 # http://www.cmake.org/Bug/view.php?id=8598
 # http://gcc.gnu.org/bugzilla/show_bug.cgi?id=35707
 # http://www.cmake.org/Bug/view.php?id=8408
-project(PARTMC Fortran C)
+if (ENABLE_TCHEM)
+  # Use the wrapper to nvcc for the CXX compiler for GPU
+  if (ENABLE_TCHEM_GPU)
+    set(ENV{CXX} $ENV{KOKKOS_PATH}/bin/nvcc_wrapper)
+    project(PARTMC Fortran C CXX)
+  else()
+    project(PARTMC Fortran C)
+  endif()
+else()
+  project(PARTMC Fortran C)
+endif()
 
 set(PACKAGE_BUGREPORT "mwest@illinois.edu")
 set(PACKAGE_NAME "PartMC")
 set(PACKAGE_STRING "PartMC 2.8.0")
 set(PACKAGE_TARNAME "partmc")
 set(PACKAGE_VERSION "2.8.0")
 
+include(FortranCInterface)
+
+# Following must be commented out if using -Wimplicit-interface with -Werror
+# Working with C and Fortran
+#FortranCInterface_VERIFY()
+#if (ENABLE_TCHEM_GPU)
+#   # Working with C++ and Fortran
+#   FortranCInterface_VERIFY(CXX)
+#endif()
+
+if (ENABLE_TCHEM)
+  set(CMAKE_C_FLAGS "-lpthread -ldl -fopenmp" CACHE STRING "" FORCE)
```

Comment:
`-fopenmp` is specific to clang/gcc, to support other compilers it would likely help to use CMake's OpenMP handling and its `OpenMP_<lang>_FLAGS` var: https://cmake.org/cmake/help/latest/module/FindOpenMP.html

Reply:
Thanks @slayoo for the suggestion. Addressed in https://github.com/compdyn/partmc/pull/198/commits/f165310e8b3159dbebddc20350409c949fad354b

Reply:
Actually, the same concerns `-lpthread` and `-ldl`, here's an example how to make it more portable:
https://github.com/alex85k/sqlite3-cmake/blob/121e7d273436c5de88956f559b940b8d69cadfbe/CMakeLists.txt#L26-L27

Reply:
(anticipating build troubles with Windows for PyPartMC :) )